tests/eas/cpuhp.py

```diff
@@ -0,0 +1,217 @@
+
+    @classmethod
+    def _experimentsInit(cls, *args, **kwargs):
+        # calibration shoud be done before starting spurious hotplugs
+        if not cls.te._calib:
+            cls.te.calibration()
+        
+        # Choose a random seed explicitly if not given
+        if 'seed' not in cls.hp_stress or cls.hp_stress['seed'] is None:
+            random.seed() 
+            cls.hp_stress['seed'] = random.randint(0, sys.maxint)
+        random.seed(cls.hp_stress['seed'])
+        cls.te._log.info('Random sequence of CPU Hotplug generated with: ')
+        cls.te._log.info(cls.hp_stress)
+        with open(os.path.join(cls.te.res_dir, 'hotplug_cfg.json'), 'w') as f:
+            json.dump(cls.hp_stress, f, sort_keys=True, indent=4)
+       
+        # Play with (online) hotpluggable CPUs only
+        cls.hotpluggable_cpus = filter(
+                lambda x: cls.target.file_exists(hp._cpu_path(cls.target, x)),
+                cls.target.list_online_cpus())
+        if not cls.hotpluggable_cpus:
+            raise RuntimeError('Cannot find any hotpluggable CPU online')
+        cls.te._log.info('Hotpluggable CPUs found on target: ')
+        cls.te._log.info(cls.hotpluggable_cpus)
+
+        # Run random hotplug sequence on target
+        cpuhp_seq_script = cls._random_cpuhp_script()
+        cmd = cls.target.install(cpuhp_seq_script)
+        cls.hotplug_shell = cls.target.background(cmd, as_root=True)
+        cls.te._log.info('Hotplug stress has now started on target')
+    
+    @classmethod
+    def _experimentsFinalize(cls, *args, **kwargs):
+        cls.te._log.info('Stopping hotplug stress on the target now')
+        # Kill hotplug script
+        pid = cls.target.read_int(cls.pid_file)
+        cls.target.kill(pid, as_root=True)
+        cls.hotplug_shell.kill() 
+        # Resume to original state
+        cls.te._log.info('Plugging back in currently offline CPUs')
+        cls.target.hotplug.online(*cls.hotpluggable_cpus)
+
+    @classmethod
+    def _random_cpuhp_script(cls):
+        '''
+        Generate a script consisting of a random sequence of hotplugs operations
+        
+        Two consecutive hotplugs can be separated by a random (and configurable 
+        through .hp_stress) sleep in the script. Each hotplug operation is
+        logged in a ftrace marker file for post processing. The return value is
+        the path (on the host) where the script was saved.
+
+        Example of generated script:
+        > #!/system/bin/sh
+        > echo $$ > /data/local/tmp/devlib-target/hotplug.pid
+        > while true; do
+        >     echo 0 > /sys/devices/system/cpu/cpu1/online
+        >     echo "Plug off CPU 1" > /sys/kernel/debug/tracing/trace_marker
+        >     sleep 0.4245
+        >     echo 1 > /sys/devices/system/cpu/cpu1/online
+        >     echo "Plug in CPU 1" > /sys/kernel/debug/tracing/trace_marker
+        >     sleep 0.178
+        > done
+        '''
+        script_path = '/tmp/random_cpuhp.sh'
```

Comment:
I think it might be better to keep the script in the results directory (which is what TargetScript does), so that it can be easily re-used/checked in case of failure. It's slightly less relevant here since with a given seed we should always get the same script, but I still think it's a good practice.